src/XRP/BinaryCoding.h

```diff
@@ -17,10 +17,19 @@ enum class FieldType;
 /// Encodes a field type.
 inline void encodeType(FieldType type, int key, std::vector<uint8_t>& data) {
     const auto typeValue = static_cast<int>(type);
-    if (key <= 0xf) {
-        data.push_back(static_cast<uint8_t>((typeValue << 4) | key));
+    if (static_cast<int>(typeValue) <= 0xf) {
+        if (key <= 0xf) {
+            data.push_back(static_cast<uint8_t>((typeValue << 4) | key));
+        } else {
+            data.push_back(static_cast<uint8_t>(typeValue << 4));
+            data.push_back(static_cast<uint8_t>(key));
+        }
+    } else if(key <= 0xf) {
+        data.push_back(static_cast<uint8_t>(key));
+        data.push_back(static_cast<uint8_t>(typeValue));
     } else {
-        data.push_back(static_cast<uint8_t>(typeValue << 4));
+        data.push_back(0);
+        data.push_back(static_cast<uint8_t>(typeValue));
         data.push_back(static_cast<uint8_t>(key));
     }
```

Comment:
Here I suggest to use `emplace_back()` instead of `push_back()`

Reply:
I'm happy to change it but isn't emplace_back() the same as push_back() for a primitive?

Reply:
It's just more idiomatic, as emplace back construct the object in place, even if there is not extra optimization for integer we reduce the number of code smells/clang tidy warnings